lib/hq/graphql/comparator.rb

```diff
@@ -0,0 +1,50 @@
+# typed: strong
+# frozen_string_literal: true
+
+require "graphql/schema_comparator"
+
+module HQ
+  module GraphQL
+    class Comparator
+      # Higher values will include changes from criticalities with lower values as well.
+      # For example, setting the criticality as dangerous will return dangerous and breaking changes.
+      CRITICALITY = {
+        breaking: 0,
+        dangerous: 1,
+        non_breaking: 2
+      }
+
+      class << self
+        def compare(old_schema, new_schema, criticality: :breaking)
```

Comment:
`::HQ::GraphQL::Comparator.compare("/app/graphql/schema.graphql")` is really what we want. We want to compare the current schema that exists in code with a checked in file. I would prefer if this function could handle creation and cleanup of the new schema file

Reply:
Aren't there possibly multiple schemas though? In agencies for example, how would it know whether to check AgencieshqGraphqlSchema or InternalGraphqlSchema :)?